### ISSUE TITLE
Vehicle reports "no valid destination" while delivering to valid destination

### ISSUE THREAD

**Issue by super-aardvark** [MaFi-Games/Captain-of-Industry-issues#628]
### Issue description

I caught one of my pickups with the "cannot deliver" warning as it drove around making deliveries.

### Expected behavior

"Cannot deliver" warning should go away when a valid destination is found.

### Reproduction steps

Unknown, and I didn't think to save at the time.  Sorry.

### Game version

v0.4.8

### Attachments

![captain_destination_bug](https://user-images.githubusercontent.com/3050938/180907433-dad31fe7-c8cb-466b-a88a-274d8f74e9c4.png)

**Comment by zuff-coi** [comment]
Could you provide a save game for this issue? Is it still happening in experimental update v0.4.9c?

**Comment by super-aardvark** [comment]
No, sorry, I didn't think to make a save at the time.  I haven't noticed the issue since my initial report.

**Comment by zuff-coi** [comment]
Glad to hear you haven't run into again, @super-aardvark . If you run into it again let us know and try to get a save file for us! Thanks!

**Comment by super-aardvark** [comment]
I just observed this (or very similar) behavior in 0.4.14a (154).  Here's what happened just before:

1. Truck had a load of water, but was stuck at the pickup point because it couldn't find a path to the destination.  Truck showed appropriate "Truck cannot deliver. No valid destination" warning.
3. I switched Truck Import to OFF on the destination machine (copper electrolysis).
4. Truck dropped its load (seems to have disappeared, rather than being dumped back into the water container, though I can't be sure) and set off on a new task with no inventory (destination is a Steel storage).
5. Truck continued to show the "Truck cannot deliver. No valid destination" warning (but the warning no longer shows in the upper-right corner of the screen).

Save file attached.

[truck-deliver-error-bug.zip](https://github.com/MaFi-Games/Captain-of-Industry-issues/files/10773709/truck-deliver-error-bug.zip)
